### ISSUE TITLE
Array query parameters

### ISSUE THREAD

**Issue by jvv** [vuejs/vue-resource#259]
To start off: I don't know if this is an issue, or setting or bug on my side.

I'm trying to create a request URL which looks like:

`/applications/6/pois/localized?max=50&category=10&category=14`

I'm creating this URL with the following data 

    url : 'applications/'+this.application.id+'/pois',
    method: 'get',
    data: {
        max: 50,
        category: this.selected_categories
    }

( `this.selected_categories` is an array with only ID's)

But I end up with a request URL which looks like:

`/applications/6/pois/localized?max=50&category%5B%5D=10&category%5B%5D=14`

Basically, I need to remove the `[]` (or: `%5B%5D`) in the uri. I tried to generate the query parameters myself, but this ends up in the `&` being rewritten.

Any tips on how to solve this?

**Comment by jvv** [comment]
Same issue as #217 ^

**Comment by steffans** [comment]
You can use [URI templates](https://medialize.github.io/URI.js/uri-template.html) for this, heres an example:
```js
url : 'applications{/id}/pois{?max,category}',
method: 'get',
data: {
    id: this.application.id,
    max: 50,
    category: this.selected_categories
}
```

**Comment by jvv** [comment]
Thanks, that did the trick.

**Comment by milashensky** [comment]
I had same problem and solve it this way. 
```
Vue.http.interceptors.push(function(request, next) {
    if (request.method === "GET") {
        let url = request.url data = Object.assign({}, request.params);
        for (let i = 0, key; key = Object.keys(data)[i]; i++)
            if (Array.isArray(data[key])) 
                data[key] = JSON.stringify(data[key])
        request.params = data;
    }
    next();
})
```
Maybe someone find it useful

**Comment by andreygalassi** [comment]
or maybe
```
Vue.http.interceptors.push(function(request, next) {
	if (request.method === "GET") {
		let data = Object.assign({}, request.params);
		for (let i = 0, key; key = Object.keys(data)[i]; i++)
			if (Array.isArray(data[key])) 
				request.url+='{?'+key+'*}'
	}
	next();
})
```
